sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/AmqpReceiveLinkProcessor.java

```diff
@@ -166,6 +169,21 @@ public void onNext(AmqpReceiveLink next) {
                         }
                     },
                     error -> {
+                        if (error instanceof AmqpException) {
+                            AmqpException amqpException = (AmqpException) error;
+                            if (amqpException.getErrorCondition() == AmqpErrorCondition.LINK_STOLEN
+                                && amqpException.getContext() != null
+                                && amqpException.getContext() instanceof LinkErrorContext) {
+                                LinkErrorContext errorContext = (LinkErrorContext) amqpException.getContext();
+                                if (currentLink != null
+                                    && !currentLink.getLinkName().equals(errorContext.getTrackingId())) {
```

Comment:
In what case a link name is the same as a tracking id?

Reply:
I thought the trackingId was an autogenerated value?

Reply:
When the link is stolen, the `trackingId` in LinkErrorContext is the link name.

```java
if (link.getRemoteProperties() != null && link.getRemoteProperties().containsKey(TRACKING_ID_PROPERTY)) {
    referenceId = link.getRemoteProperties().get(TRACKING_ID_PROPERTY).toString();
} else {
    referenceId = link.getName();
}
return new LinkErrorContext(getHostname(), entityPath, referenceId, link.getCredit());
```